packages/react-integration/demo-app-ts/src/components/demos/DataToolbarDemo/DataToolbarDemo.tsx

```diff
@@ -67,35 +69,37 @@ export class DataToolbarDemo extends React.Component<DataToolbarProps, DataToolb
     }));
   };
 
-  onInputChange = newValue => {
+  onInputChange = (newValue: string) => {
     this.setState({ inputValue: newValue });
   };
 
-  onSelect = (type, event, selection) => {
+  onSelect = (type: string, event, selection: string | SelectOptionObject) => {
     const checked = event.target.checked;
     this.setState(prevState => {
       const prevSelections = prevState.filters[type];
       return {
         filters: {
           ...prevState.filters,
-          [type]: checked ? [...prevSelections, selection] : prevSelections.filter(value => value !== selection)
+          [type]: checked
+            ? [...prevSelections, selection.toString()]
+            : prevSelections.filter((value: string) => value !== selection.toString())
         }
       };
     });
   };
 
-  onStatusSelect = (event, selection) => {
+  onStatusSelect = (event: React.MouseEvent | React.ChangeEvent, selection: string | SelectOptionObject) => {
     this.onSelect('status', event, selection);
   };
 
-  onRiskSelect = (event, selection) => {
+  onRiskSelect = (event: React.MouseEvent | React.ChangeEvent, selection: string | SelectOptionObject) => {
     this.onSelect('risk', event, selection);
   };
 
   onDelete = (type = '', id = '') => {
     if (type) {
       this.setState(prevState => {
-        prevState.filters[type.toLowerCase()] = prevState.filters[type.toLowerCase()].filter(s => s !== id);
+        prevState.filters[type.toLowerCase()] = prevState.filters[type.toLowerCase()].filter((s: string) => s !== id);
```

Comment:
I think `prevState` is supposed to be readonly? You could try something like

`const newState = Object.assign(prevState);` and then replace references to prevState with newState like

```js
newState.filters[type.toLowerCase()] = newState.filters[type.toLowerCase()].filter((s: string) => s !== id);
return {
  filters: newState.filters
};
```

I think if you do this for both `onDelete` and `onSelect` then we'll no longer need the `[key: string]: string[];` entry in the `Filter` interface.

Reply:
Updated onDelete in this file and in core demo, onSelect should be good as it references prevState but doesn't change it like onDelete was doing.